crates/sui-replay-2/src/execution.rs

```diff
@@ -151,15 +152,32 @@ fn get_trace_output_path(trace_execution: Option<String>) -> Result<PathBuf, Rep
     }
 }
 
-/// Saves the trace and additional metadata needed needed to analyze the trace.
+/// Saves the trace and additional metadata needed needed to analyze the trace
+/// to a subderectory named after the transaction digest.
 fn save_trace_output(
     output_path: &PathBuf,
+    digest: TransactionDigest,
     trace_builder: MoveTraceBuilder,
     env: &ReplayEnvironment,
 ) -> Result<(), ReplayError> {
+    let txn_output_path = output_path.join(digest.to_string());
+    if txn_output_path.exists() {
+        return Err(ReplayError::TracingError {
+            err: format!(
+                "Trace output directory already for transaction {} already exists: {:?}",
+                digest, txn_output_path
+            ),
+        });
+    }
+    fs::create_dir(&txn_output_path).map_err(|e| ReplayError::TracingError {
```

Comment:
Nit: probably better to use `create_dir_all` here instead in case `output_path` (and/or parents of that dir) don't also exist as well.

Reply:
The calling current calling sequence guarantees that `output_path` exists but it certainly won't hurt to use `create_dir_all` in case that changes in the future